Src/IdiomsUnitTest/GuardClauseAssertionTest.cs

```diff
@@ -1626,37 +1637,49 @@ public NestedGenericParameterTestType(Func<T1, IEnumerable<IEnumerable<T2>>, T1[
 
         private class NonProperlyGuardedClass
         {
-            private const string InvalidParamName = "invalidParamName";
-
             public NonProperlyGuardedClass(object argument)
             {
-                if (argument == null) throw new ArgumentNullException(InvalidParamName);
+                if (argument == null) throw new ArgumentNullException("INVALID");
             }
 
             public object Property
             {
                 get => null;
                 set
                 {
-                    if (value == null) throw new ArgumentNullException(InvalidParamName);
+                    if (value == null) throw new ArgumentNullException("INVALID");
                 }
             }
 
             public void Method(object argument)
             {
-                if (argument == null) throw new ArgumentNullException(InvalidParamName);
+                if (argument == null) throw new ArgumentNullException("INVALID");
```

Comment:
Instead of just "INVALID" perhaps [a descriptive message is warranted](http://blog.ploeh.dk/2014/12/23/exception-messages-are-for-programmers/)?